src/main/java/org/example/Main.java

```diff
@@ -1,7 +1,197 @@
+    private static String dateChecker() {
+        final var today = LocalDate.now();
+        String input;
+        LocalDate date;
+
+        do try {
+            input = scan.nextLine();
+            date = LocalDate.parse(input);
+            if (!date.isBefore(today)) {//date must be at least today if so loop end
+                break;
+            }
+            System.out.printf("%sThis date already was!!%s\n", RED_BOLD_BRIGHT, RESET);
+            System.out.printf("\n%sWrite date which is at least %sToday:%s ", BLUE_BOLD_BRIGHT, RED_BOLD_BRIGHT, RESET);
+        } catch (DateTimeParseException e) {
+            System.out.printf("%sInvalid date format!! %s\n", RED_BOLD_BRIGHT, RESET);
+            System.out.printf("\n%sWrite date in the format %sYYY-MM-DD%s:%s ", BLUE_BOLD_BRIGHT, YELLOW_BOLD_BRIGHT, BLUE_BOLD_BRIGHT, RESET);
+        }
+        while (true);
+        return input;
+    }
+
+    private static void removeTask() {
+        String input;
+
+        System.out.printf("\n%sWrite the number of the task you want to remove from the list%s\n", BLUE_BOLD_BRIGHT, RESET);
+        System.out.printf("%sif you don't know the number of the task write \"%s?%s\" or \"%sexit%s\" to return to the main menu:%s", BLUE_BOLD_BRIGHT, YELLOW_BOLD_BRIGHT, BLUE_BOLD_BRIGHT, YELLOW_BRIGHT, BLUE_BOLD_BRIGHT, RESET);
+        do {
+            input = scan.nextLine().toLowerCase();
+            if (NumberUtils.isDigits(input)) {
+                int num = Integer.parseInt(input);//
+                if (num >= 1 && num <= tasks.length) {
+                    tasks = ArrayUtils.remove(tasks, num - 1);//real index is from 0 but user see tasks from 1
+                    System.out.printf("Task N.%d deleted successfully", num);
+                    break;
+                }
+                System.out.printf("In the list isn't task number %d, please try again:", num);
+                continue;
+            }
+            if (input.equals("?")) {
+                showTasks();
+                System.out.printf("%sWrite the number of tasks you want to remove from the list:%s ", BLUE_BOLD_BRIGHT, RESET);
+                continue;
+            }
+            if (input.equals("exit")) break;
+            System.out.println("Invalid input!");
+            System.out.print("if you don't know the number of tasks write \"?\" or \"exit\" to return to the main menu: :");
+        } while (true);
+    }
+
+    private static void saveTask() {
+        final Path pathFile = Paths.get("tasks.csv");
+
+        if (Files.exists(pathFile)) {
+            final var data = new StringBuilder();
+            for (String[] task : tasks) {
+                for (int i = 0; i < task.length - 1; i++) {
+                    data.append(task[i]).append(",");
+                }
+                data.append(task[task.length - 1]).append(System.lineSeparator());//after last part shouldn't be "," but only lineSeparator
+            }
```

Comment:
IMPROVE: vytvoreni finalniho retezce pro ulozeni do souboru lze zjednodusit.

```suggestion
 for (var task : tasks) {
            sb.append(String.join(", ", task));
            sb.append(System.lineSeparator());
        }
```

Reply:
tady se omlouvám, ale toto zjednoduseni nelze máš tam chybu.
toto rozbije csv soubor protoze to udela toto: "popis souboru, novy radek, dastum, novy radek atd..
ja jsem oddelil ten posledni append "true/false" za tu smycku proto  ze predtim kdyz jsem mel uvnitr jeden append a za smyckou pouze lineseparator tak to sice fungovalo
ale v soboru to dělalo  value1,value2,value3,
a podle standartu
za posledni hodnotou neni comma
https://en.wikipedia.org/wiki/Comma-separated_values
takže proto ve smyčce proběhne jenom hodnota1 a 2 a třetí se přidá za smyčkou uz bez ,
a přidá se lineseparator